atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/ClientLogEvents.java

```diff
@@ -73,28 +79,23 @@ default CommitUpdate toCommitUpdate(LockWatchVersion startVersion, CommitInfo co
             return ImmutableInvalidateAll.builder().build();
         }
 
+        // We want to ensure that we do not miss any versions, but we do not care about the event with the same version
+        // as the start version.
         verifyReturnedEventsEnclosesTransactionVersions(
-                Range.closed(startVersion.version(), commitInfo.commitVersion().version()), true);
+                startVersion.version() + 1, commitInfo.commitVersion().version());
 
         LockEventVisitor eventVisitor = new LockEventVisitor(commitInfo.commitLockToken());
         Set<LockDescriptor> locksTakenOut = new HashSet<>();
         events().events().forEach(event -> locksTakenOut.addAll(event.accept(eventVisitor)));
         return ImmutableInvalidateSome.builder().invalidatedLocks(locksTakenOut).build();
     }
 
-    default void verifyReturnedEventsEnclosesTransactionVersions(Range<Long> versionRange, boolean offsetStartVersion) {
-        // If we offset the start version, but the range is already [x..x], we throw when creating the range.
-        if (versionRange.lowerEndpoint().equals(versionRange.upperEndpoint()) && offsetStartVersion) {
+    default void verifyReturnedEventsEnclosesTransactionVersions(long lowerBound, long upperBound) {
+        if (lowerBound > upperBound) {
             return;
         }
 
-        Range<Long> rangeToTest;
-        if (offsetStartVersion) {
-            rangeToTest = Range.closed(versionRange.lowerEndpoint() + 1, versionRange.upperEndpoint());
-        } else {
-            rangeToTest = versionRange;
-        }
-
+        Range<Long> rangeToTest = Range.closed(lowerBound, upperBound);
```

Comment:
`Range.openClosed` so you don't need to do +1 on the lower bound everywhere

Reply:
Though I guess you need a -1 :D

Reply:
Yeah nope, because there is one case where you need something different - probably easier at this point to just add the one.